app/effector.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+import json
+import os
+
+import spade
+from spade.agent import Agent
+from spade.behaviour import State
+
+
+class EffectorAgent(Agent):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    async def setup(self):
+        print(f"Effector agent with id: {self.jid} initialized")
+        b = self.FlipFlop()
+        template = spade.template.Template()
+        template.set_metadata("performative", "request")
+        self.add_behaviour(b, template)
+
+    class FlipFlop(State):
+        """responds to request with agree/cancel and state in msg.body"""
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+        def perform_internal_action(self, state):
+            pass
+
+        def change_state(self, state):
+            self.set_next_state(state)
+            self.perform_internal_action(state)
+            print(f"Effector agent changed state to: {self.next_state}")  
+
+        async def run(self):
+            msg = await self.receive()
+            if msg:
+                print(f"Effector agent received request: {msg}")
+                if msg.body != self.next_state:
+                    self.change_state(msg.body)
+                    reply = msg.make_reply()
```

Comment:
Czy testowałeś ten kod?

Reply:
Pytam, bo nie jestem pewien, czy typ State będzie działać przy twojej implementacji.

https://spade-mas.readthedocs.io/en/latest/behaviours.html#finite-state-machine-behaviour

W jaki sposób będzie triggerowany  ten twój State?

Reply:
Założyłem że state może być czymkolwiek np. stringiem. Służy tylko jako informacja jaki jest stan. Nie jest to rzeczywiście state w rozumieniu biblioteki. Funkcja set_next_state przypisuje tylko state do self.next_state, więc kod powinien działać.